### PR TITLE
Update to pact 4.7.1

### PR DESCRIPTION
**Pull request by rsoeldner**
This PR is now superseded by #1661 

# TODO
don't merge this until
 -  [x] pact-4.7.1 release is merged 
 -  [ ] the pointers to pact have been updated in this PR.

---

- Bumps pact version to 4.7.1
- Removes runtime return value typechecking tests (typechecking is moved under a standalone flag (not in the `DisablePact**` family) in Pact so that chainweb may manually disable the checks.

### CODE REVIEW

**Review by larskuhtz — APPROVED** [review]

File: src/Chainweb/Pact/TransactionExec.hs [hunk 1]
```diff
@@ -304,6 +305,7 @@ applyGenesisCmd logger dbEnv spv cmd =
           , FlagDisablePact43
           , FlagDisablePact44
           , FlagDisablePact45
+          , FlagDisableRuntimeReturnTypeChecking
```
Comment (sirlensalot) [inline src/Chainweb/Pact/TransactionExec.hs:308]: Pretty sure this isn't necessary. PactMultiChainTest would catch it if so

File: src/Chainweb/Pact/TransactionExec.hs [hunk 2]
```diff
@@ -372,7 +374,8 @@ applyCoinbase v logger dbEnv (Miner mid mks@(MinerKeys mk)) reward@(ParsedDecima
       enablePact431 txCtx ++
       enablePact44 txCtx ++
       enablePact45 txCtx ++
-      enablePact47 txCtx
+      enablePact47 txCtx ++
+      [ FlagDisableRuntimeReturnTypeChecking ]
```
Comment (sirlensalot) [inline src/Chainweb/Pact/TransactionExec.hs:378]: probably unnecessary

File: src/Chainweb/Pact/TransactionExec.hs [hunk 3]
```diff
@@ -578,12 +581,12 @@ applyUpgrades v cid height
   where
     installCoinModuleAdmin = set (evalCapabilities . capModuleAdmin) $ S.singleton (ModuleName "coin" Nothing)
 
-    applyCoinV2 = applyTxs (upgradeTransactions v cid) [FlagDisableInlineMemCheck, FlagDisablePact43, FlagDisablePact45]
+    applyCoinV2 = applyTxs (upgradeTransactions v cid) [FlagDisableInlineMemCheck, FlagDisablePact43, FlagDisablePact45, FlagDisableRuntimeReturnTypeChecking]
```
Comment (sirlensalot) [inline src/Chainweb/Pact/TransactionExec.hs:584]: I would guess everything in `applyUpgrades` doesn't need the flag, but PactMultiChainTest would find it if it was necessary.

File: test/Chainweb/Test/Pact/PactMultiChainTest.hs [hunk 4]
```diff
@@ -857,10 +857,6 @@ chainweb219UpgradeTest = do
         "Should not resolve new pact native: dec"
         -- Should be cannot resolve dec, but no errors pre-fork.
         ""
-      , PactTxTest runIllTypedFunction $
```
Comment (sirlensalot) [inline test/Chainweb/Test/Pact/PactMultiChainTest.hs:860]: We actually need this coverage pre- and post-fork

**Review by sirlensalot — APPROVED** [review]
This can merge as-is, so my comments are to be addressed if we feel we have time, or we can ship this in interests of getting 2.19.1 out. 

However, if we do merge this, we need to verify in manual/JS testing that the return RTC is actually disabled in `local` and in block production post-fork.

**Comment by larskuhtz** [comment]
This PR is superseded by https://github.com/kadena-io/chainweb-node/pull/1661

**Pull request closed without merging**